test-poc/framework/src/main/java/org/keycloak/test/framework/config/Config.java

```diff
@@ -1,54 +1,39 @@
 package org.keycloak.test.framework.config;
 
+import io.smallrye.config.DotEnvConfigSourceProvider;
+import io.smallrye.config.SmallRyeConfig;
+import io.smallrye.config.SmallRyeConfigBuilder;
+import org.eclipse.microprofile.config.spi.ConfigSource;
 import org.keycloak.test.framework.injection.ValueTypeAlias;
 
-import java.io.File;
-import java.io.FileInputStream;
-import java.io.IOException;
-import java.util.Properties;
+import java.util.ArrayList;
+import java.util.List;
 
 public class Config {
 
-    private final static Config instance = new Config();
+    private static final SmallRyeConfig config;
 
-    private Properties localEnv = new Properties();
+    static {
+        List<ConfigSource> configSources = new ArrayList<>();
 
-    private Config() {
-        File envFile = new File(".env");
-        if (envFile.isFile()) {
-            try {
-                localEnv.load(new FileInputStream(envFile));
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
+        String kcTestConfigPath = System.getProperty("kc.test.config");
```

Comment:
Don't you want to create private static final String testConfig = "kc.test.config" variable in here for better readability?